tests/main/hook-permissions/task.yaml

```diff
@@ -0,0 +1,44 @@
+summary: Ensure that post-refresh hook has permissions for its connected slot.
+
+# Note: it is critical for this test to use a slot that requires
+# peer=<plug security tag> expression label for dbus, and not simply
+# label=unconfined on a permanent slot snippet.
+details: |
+    Ensure that post-refresh hook has permissions to query upower-observe slot
+    and enumerate upower devices.
+
+systems: [-fedora-*, -opensuse-*, -arch-*, -amazon-*, -centos-*]
+
+restore: |
+    rm -f ./enumerate.txt
+    snap remove --purge test-snap || true
+    # shellcheck source=tests/lib/systems.sh
+    . "$TESTSLIB/systems.sh"
+    if is_core_system; then
+      snap remove --purge upower || true
+    fi
+
+prepare: |
+    # shellcheck source=tests/lib/systems.sh
+    . "$TESTSLIB/systems.sh"
+    if is_core_system; then
+        snap install upower
+    fi
+    snap pack test-snap
```

Comment:
you could use install_local function here, it looks first in the local directory to find the snap and otherwise it goes to $TESTLIB/snaps and search there. It is going to use the snap file and install it and it is doesn't exist it is going to pack and install it.

Reply:
Right, thanks, updated.